### PR TITLE
 fix(components): [tabs] fix watch only trigger once

### LINKED ISSUE #9664: [Component] [tabs] tabs的tab-click、tab-change设置后不能禁止修改model-value
Bug Type: **`Component`**

## Environment

- Vue Version: `3.2.38`
- Element Plus Version: `2.2.16`
- Browser / OS: `c`
- Build Tool: `Vite`

### Related Component

- `el-tabs`

### Reproduction Link

[Element Plus Playground](https://element-plus.run/#eyJBcHAudnVlIjoiPHNjcmlwdCBzZXR1cCBsYW5nPVwidHNcIj5cbmltcG9ydCB7IHJlZiwgdmVyc2lvbiBhcyB2dWVWZXJzaW9uIH0gZnJvbSAndnVlJ1xuaW1wb3J0IHsgdmVyc2lvbiBhcyBFcFZlcnNpb24gfSBmcm9tICdlbGVtZW50LXBsdXMnXG5pbXBvcnQgeyBFbGVtZW50UGx1cyB9IGZyb20gJ0BlbGVtZW50LXBsdXMvaWNvbnMtdnVlJ1xuXG5jb25zdCBhY3RpdmVOYW1lID0gcmVmKCdmaXJzdCcpXG5cbmNvbnN0IGhhbmRsZUNsaWNrID0gKHRhYjogVGFic1BhbmVDb250ZXh0LCBldmVudDogRXZlbnQpID0+IHtcbiAgYWN0aXZlTmFtZS52YWx1ZT0ndGhpcmQnXG59XG5jb25zdCBoYW5kbGVDaGFuZ2UgPSAobmFtZSkgPT4ge1xuICBhY3RpdmVOYW1lLnZhbHVlID0gJ3RoaXJkJ1xufVxuPC9zY3JpcHQ+XG5cbjx0ZW1wbGF0ZT5cbiAgPGVsLXRhYnMgOm1vZGVsLXZhbHVlPVwiYWN0aXZlTmFtZVwiIGNsYXNzPVwiZGVtby10YWJzXCIgQHRhYi1jbGljaz1cImhhbmRsZUNsaWNrXCIgQHRhYi1jaGFuZ2U9XCJoYW5kbGVDaGFuZ2VcIj5cbiAgICA8ZWwtdGFiLXBhbmUgbGFiZWw9XCJVc2VyXCIgbmFtZT1cImZpcnN0XCI+VXNlcjwvZWwtdGFiLXBhbmU+XG4gICAgPGVsLXRhYi1wYW5lIGxhYmVsPVwiQ29uZmlnXCIgbmFtZT1cInNlY29uZFwiPkNvbmZpZzwvZWwtdGFiLXBhbmU+XG4gICAgPGVsLXRhYi1wYW5lIGxhYmVsPVwiUm9sZVwiIG5hbWU9XCJ0aGlyZFwiPlJvbGU8L2VsLXRhYi1wYW5lPlxuICAgIDxlbC10YWItcGFuZSBsYWJlbD1cIlRhc2tcIiBuYW1lPVwiZm91cnRoXCI+VGFzazwvZWwtdGFiLXBhbmU+XG4gIDwvZWwtdGFicz5cbjwvdGVtcGxhdGU+XG4iLCJpbXBvcnRfbWFwLmpzb24iOiJ7XG4gIFwiaW1wb3J0c1wiOiB7fVxufSIsIl9vIjp7fX0=)

### Steps to reproduce

点击tab不能固定选项

### What is Expected?

固定选项

### What is actually happening?

切换到点击的项

### Additional comments

_(empty)_

### PR DESCRIPTION
**Pull request by faga295**
close #9664 

Please make sure these boxes are checked before submitting your PR, thank you!

- [x] Make sure you follow contributing guide [English](https://github.com/element-plus/element-plus/blob/master/.github/CONTRIBUTING.en-US.md) | ([中文](https://github.com/element-plus/element-plus/blob/master/.github/CONTRIBUTING.zh-CN.md) | [Español](https://github.com/element-plus/element-plus/blob/master/.github/CONTRIBUTING.es.md) | [Français](https://github.com/element-plus/element-plus/blob/master/.github/CONTRIBUTING.fr-FR.md)).
- [x] Make sure you are merging your commits to `dev` branch.
- [x] Add some descriptions and refer to relative issues for your PR.

### CODE REVIEW

File: packages/components/tabs/src/tabs.tsx [hunk 1]
```diff
@@ -87,13 +87,17 @@ export default defineComponent({
...
-    const currentName = ref<TabPanelName>(
-      props.modelValue ?? props.activeName ?? '0'
-    )
+    const currentName = computed<TabPanelName>({
+      get() {
+        return props.modelValue ?? props.activeName ?? '0'
+      },
+      set(value) {
+        emit(UPDATE_MODEL_EVENT, value)
+      },
```
Comment (tolking) [inline packages/components/tabs/src/tabs.tsx:96]: The `modelValue` and ~~`activeName`~~ attribute should be Control Props.

Should respond to value modifications outside the component, avoid modifications inside the component

Comment (faga1) [reply packages/components/tabs/src/tabs.tsx:96]: I emit update:model-value,it seems like respond to outside the component,how should I respond correctly

File: packages/components/tabs/src/tabs.tsx [hunk 2]
```diff
@@ -87,13 +87,17 @@ export default defineComponent({
...
+    })
 
     const changeCurrentName = (value: TabPanelName) => {
       currentName.value = value
```
Comment (tolking) [inline packages/components/tabs/src/tabs.tsx:100]: I think the problem is here, the currentName value should not be modified inside the component

File: packages/components/tabs/src/tabs.tsx (hunk 1 above)
Comment (tolking) [reply packages/components/tabs/src/tabs.tsx:96]: The `set` attribute in the computed is redundant.

File: packages/components/tabs/src/tabs.tsx (hunk 2 above)
Comment (faga1) [reply packages/components/tabs/src/tabs.tsx]: if we use dynamic props instead of v-model,the tab will not switch when we click nav

Comment (tolking) [reply packages/components/tabs/src/tabs.tsx]: test failed

Comment (faga1) [reply packages/components/tabs/src/tabs.tsx]: sorry,this issue is out of my depth,I will close this pr,thank you

**Pull request closed without merging**